command/format.go

```diff
@@ -0,0 +1,160 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	hclutils "github.com/hashicorp/packer/hcl2template"
+	"github.com/posener/complete"
+)
+
+const (
+	hcl2FileExt        = ".pkr.hcl"
+	hcl2JsonFileExt    = ".pkr.json"
+	hcl2VarFileExt     = ".auto.pkrvars.hcl"
+	hcl2VarJsonFileExt = ".auto.pkrvars.json"
+)
+
+type FormatCommand struct {
+	Meta
+	parser *hclparse.Parser
+}
+
+func (c *FormatCommand) Run(args []string) int {
+	ctx := context.Background()
+	cfg, ret := c.ParseArgs(args)
+	if ret != 0 {
+		return ret
+	}
+
+	// Might need to check for cfg type.
+	return c.RunContext(ctx, cfg)
+}
+
+func (c *FormatCommand) ParseArgs(args []string) (*FormatArgs, int) {
+	var cfg FormatArgs
+	flags := c.Meta.FlagSet("format", FlagSetNone)
+	flags.Usage = func() { c.Ui.Say(c.Help()) }
+	cfg.AddFlagSets(flags)
+	if err := flags.Parse(args); err != nil {
+		return &cfg, 1
+	}
+
+	// What is Path here?
+	args = flags.Args()
+	if len(args) == 1 {
+		cfg.Path = args[0]
+	}
+
+	return &cfg, 0
+}
+
+func (c *FormatCommand) RunContext(ctx context.Context, cla *FormatArgs) int {
+	hclFiles, _, diags := hclutils.GetHCL2Files(cla.Path, hcl2FileExt, hcl2JsonFileExt)
+	ret := writeDiags(c.Ui, nil, diags)
+	if ret != 0 {
+		return ret
+	}
+
+	hclVarFiles, _, diags := hclutils.GetHCL2Files(cla.Path, hcl2VarFileExt, hcl2VarJsonFileExt)
+	ret = writeDiags(c.Ui, nil, diags)
+	if ret != 0 {
+		return ret
+	}
+
+	allHclFiles := append(hclFiles, hclVarFiles...)
+
+	if len(allHclFiles) == 0 {
+		c.Ui.Say("No HCL files found; please check that all HCL files end with the proper suffix")
+		return 0
+	}
+
+	c.parser = hclparse.NewParser()
+	for _, path := range allHclFiles {
+		if err := c.formatFile(path, cla.Write); err != nil {
+			c.Ui.Say(err.Error())
+			return 1
+		}
+	}
```

Comment:
I think that if there is more than one file we should say which file we are formatting here.

Reply:
I had the same thought but found the names getting lost in the output. I spent a little time looking into `terraform fmt` and fount that the fmt command doesn't show the new changes by default just the file names of files to be modified which is a nice touch. The latter with the diff flag makes for a combination. I'll update.